documentation/src/main/asciidoc/topics/proc_implementing_custom_merge_policies.adoc

```diff
@@ -0,0 +1,53 @@
+[id="implementing-custom-merge-policies_{context}"]
+= Implementing custom merge policies
+
+You can create implementations of the `EntryMergePolicy` API to use custom merge policies to handle network partitions.
+
+.Procedure
+
+. Create an implementation of the `EntryMergePolicy` API.
++
+[source,java,options="nowrap",subs=attributes+]
+----
+include::code_examples/CustomMergePolicyEntryMerge.java[]
+----
+.Configure your cache to use the custom merge policy, for example:
++
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/custom_merge_policy.xml[]
+----
++
+.ConfigurationBuilder
+[source,java,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::code_examples/CustomMergePolicy.java[]
+----
+
+.Next steps
+
+For remote caches, you need to deploy your merge policy implementation to {brandname} Server.
+Complete the following steps:
+
+. Package your classes as a JAR file that includes a `META-INF/services/org.infinispan.conflict.EntryMergePolicy` file that contains the fully qualified class name of your merge policy.
++
+----
+# List implementations of EntryMergePolicy with the full qualified class name
+org.example.CustomMergePolicy
+----
+. Add the JAR file to the `server/lib` directory.
++
+[TIP]
+====
+Use the `install` command with the {brandname} Command Line Interface (CLI) to download the JAR to the `server/lib` directory.
+====
+
+//Review question: Is this bit still relevant or out of date?
+In order for a Custom merge policy to be utilised on the server, you should enable object storage, if your policies semantics require access to the stored Key/Value objects. This is because cache entries in the server may be stored in a marshalled format and the Key/Value objects returned to your policy would be instances of WrappedByteArray.
+However, if the custom policy only depends on the metadata associated with a cache entry, then object storage is not required and should be avoided (unless needed for other reasons) due to the additional performance cost of marshalling data per request.
+Finally, object storage is never required if one of the provided merge policies is used.
```

Comment:
@ryanemerson I realize you're on PTO as I write this but this bit of the doc seems out of date. Can you please review and provide some guidance? Should we update this to be more relevant to Protobuf?

Reply:
If the user has configured `application/x-protostream` as the content type then the key/value returned to the EntryMergePolicy will be a `byte[]`, which is fine if the user only wants to use the metadata for merge comparison. However, if the object is required than the user still needs to configure `application/x-java-object` as the content type on the server. If the clients are providing `application/x-protostream` bytes, then the relevant protostream marshallers will need to be deployed on the server so that the byte -> object transformation can happen.